torch/distributions.py

```diff
@@ -108,6 +120,9 @@ def __init__(self, probs):
     def sample(self):
         return torch.multinomial(self.probs, 1, True).squeeze(-1)
 
+    def sample_n(self, n):
+        return torch.cat([self.sample().unsqueeze(0) for _ in range(n)], dim=0)
```

Comment:
I think this can be done by simply calling the `torch.multinomail` function with `num_samples=n` and transposing i.e.
```python
return torch.multinomial(self.probs, n, True).t()
```

Reply:
It turns out that when `.t()` is applied to a 1D Tensor, the operation fails. Your proposed method more efficient though, so I kept the change for 2D and added a conditional to catch the 1D case with `self.sample.expand(1, 1)` instead.